core/src/apps/bitcoin/get_public_key.py

```diff
@@ -2,21 +2,33 @@
 
 from trezor import wire
 from trezor.enums import InputScriptType
-from trezor.messages import HDNodeType, PublicKey
+from trezor.messages import HDNodeType, PublicKey, UnlockPath
 
 from apps.common import coininfo, paths
-from apps.common.keychain import get_keychain
+from apps.common.keychain import FORBIDDEN_KEY_PATH, get_keychain
 
 if TYPE_CHECKING:
     from trezor.messages import GetPublicKey
+    from trezor.protobuf import MessageType
 
 
-async def get_public_key(ctx: wire.Context, msg: GetPublicKey) -> PublicKey:
+async def get_public_key(
+    ctx: wire.Context, msg: GetPublicKey, auth_msg: MessageType | None = None
+) -> PublicKey:
     coin_name = msg.coin_name or "Bitcoin"
     script_type = msg.script_type or InputScriptType.SPENDADDRESS
     coin = coininfo.by_name(coin_name)
     curve_name = msg.ecdsa_curve_name or coin.curve_name
 
+    if msg.address_n and msg.address_n[0] == paths.SLIP25_PURPOSE:
+        # UnlockPath is required to access SLIP25 paths.
+        if auth_msg is None or not UnlockPath.is_type_of(auth_msg):
```

Comment:
i believe `is_type_of(None)` would correctly return False, but maybe the typechecker won't allow that? (which might be fixable by allowing an argument of type `Any` to `is_type_of`)

Reply:
> i believe `is_type_of(None)` would correctly return False,

Yes, done in https://github.com/trezor/trezor-firmware/pull/2289/commits/aafaadabaedac2e0f35e98a07985fb782bbd19b0.

> which might be fixable by allowing an argument of type `Any` to `is_type_of`)

Done in https://github.com/trezor/trezor-firmware/pull/2289/commits/1eb3c7f945ed31224ab31ab0dbd000714a536769.